app/lua/lobject.h

```diff
@@ -340,13 +203,7 @@ typedef TValuefields TValue;
 #define setobj2n	setobj
 #define setsvalue2n	setsvalue
 
-#ifndef LUA_PACK_VALUE
-#define setttype(obj, tt) (ttype(obj) = (tt))
-#else // #ifndef LUA_PACK_VALUE
-/* considering it used only in lgc to set LUA_TDEADKEY */
-/* we could define it this way */
-#define setttype(obj, _tt) ( ttype_sig(obj) = add_sig(_tt) )
-#endif // #ifndef LUA_PACK_VALUE
+#define setttype(obj, stt) ((void) (obj)->value, (obj)->tt = (stt))
```

Comment:
Hmm? Are you trying to silence some compiler warning here or something?

Reply:
There are two `tt` fields, one in the TValue and (in the case of GCObjects) one in the GCO header, and the code should confuse the two.  (Lua 5.3 does this by renaming the latter  to `tt_`.)  The `ttype(o)` and `setttype(o,t)` macros should only be used on the TValue type.  

The `(void) (obj)->value` guard is compiled but discarded during code optimisation, but it still throws a compile error is these are used on the GCOs, and this allowed me to fix a few cases that had been missed in the eLua changes.

The reason that this is important for LFS is that the TValue form is a 32 bit integer and so testing this doesn't throw a `l8ui` exception.  What I intend to do is to replace the relatively small number of GCO `tt` tests by a macro form which generates the extra `extui` instruction and avoids the unaligned exception.

Reply:
Aaaah, that makes sense. Cool hack. Maybe worth leaving a comment along those lines there for future reference? E.g. `// guard against confusion between GCO's and TValue's "tt" field by referencing the "value" member too`

Reply:
In fact I just copied the same hack from somewhere else.  I should really back it out or make it a `//#define` variant because I really only need this when i need to do type checking in macros and I suspect that it does leave the code in in `-O0`.